dashboard/app/models/user.rb

```diff
@@ -50,10 +50,13 @@
 #  races                    :string(255)
 #  primary_contact_info_id  :integer
 #  unlock_token             :string(255)
+#  cap_state                :string(1)
+#  cap_state_date           :datetime
```

Comment:
Could you please give some detail about why you chose to create these columns of al the properties? Can you give an example of how they can be used to delete a locked-out user?

Reply:
https://github.com/code-dot-org/code-dot-org/blob/95eead1b0dd4656e003f96203517e58fd2bb48af/dashboard/lib/queries/child_account.rb#L13-L16

Reply:
How long would it take to iterate through all users by querying the user object such as in:
`User.where("cap_state = 'l'").find_each`?

Reply:
Since the `cap_state` column is indexed, the query shouldn't take longer than a few milliseconds

Reply:
Since the `User` model uses the Paranoia gem, which adds a filter by the `deleted_at` column, the only issue could arise if the MySQL query optimizer decides to use `index_users_on_deleted_at` instead of `index_users_on_cap_state_and_cap_state_date`. This can be fixed in two ways: either exclude the filter by the `deleted_at` column from the query or force the query optimizer to use the required index (not a good solution).

Reply:
We don't expect a large number of users to be affected initially, as this will only impact Colorado students under 13 for 7 days. However, the number of locked-out users could increase significantly in the future as we add more US states.